silk/datadog_checks/silk/check.py

```diff
@@ -0,0 +1,224 @@
+        super(SilkCheck, self).__init__(name, init_config, instances)
+
+        server = self.instance.get("host_address")
+        self.latest_event_query = int(get_timestamp())
+        self.url = "{}/api/v2/".format(server)
+
+        host = urlparse(server).netloc
+        self._tags = self.instance.get("tags", []) + ["silk_host:{}".format(host)]
+
+        if self.instance.get("enable_read_write_statistics"):
+            self.METRICS_TO_COLLECT.update(READ_WRITE_METRICS)
+
+        if self.instance.get("enable_blocksize_statistics"):
+            self.METRICS_TO_COLLECT.update(BLOCKSIZE_METRICS)
+
+        # System tags are collected from the /state/endpoint
+        self._system_tags = []
+
+    def check(self, _):
+        # Get system state and tags
+        self.submit_system_state()
+        # Submit service checks for each server
+        self.submit_server_state()
+
+        metric_tags = self._tags + self._system_tags
+
+        # Get events
+        self.collect_events(metric_tags)
+
+        # Get metrics
+        get_method = getattr
+
+        for path, metrics_obj in self.METRICS_TO_COLLECT.items():
+            # Need to submit an object of relevant tags
+            try:
+                response_json, _ = self._get_data(path)
+                self.parse_metrics(
+                    response_json, path, metrics_mapping=metrics_obj, get_method=get_method, tags=metric_tags
+                )
+            except Exception as e:
+                self.log.debug("Encountered error getting Silk metrics for path %s: %s", path, str(e))
+        self.service_check(self.CONNECT_SERVICE_CHECK, AgentCheck.OK, tags=self._tags)
+
+    def submit_system_state(self):
+        # Get Silk State
+        try:
+            response_hits, code = self._get_data(STATE_ENDPOINT)
+        except Exception as e:
+            self.warning("Encountered error getting Silk state: %s", str(e))
+            self.service_check(self.CONNECT_SERVICE_CHECK, AgentCheck.CRITICAL, message=str(e), tags=self._tags)
+            self.service_check(self.STATE_SERVICE_CHECK, AgentCheck.UNKNOWN, message=str(e), tags=self._tags)
+            raise
+        else:
+            if response_hits:
+                data = response_hits[0]
+                state = data.get('state').lower()
+
+                # Assign system-wide tags and metadata
+                self._assign_host_tags(data)
+                self._submit_version_metadata(data.get('system_version'))
+                self.service_check(self.STATE_SERVICE_CHECK, STATE_MAP[state], tags=self._tags)
+            else:
+                msg = "Could not access system state, got response: {}".format(code)
+                self.log.debug(msg)
+
+    def submit_server_state(self):
+        # Get Silk State
+        try:
+            server_data, code = self._get_data(SERVERS_ENDPOINT)
+        except Exception as e:
+            self.warning("Encountered error getting Silk state: %s", str(e))
```

Comment:
Can this warning message be updated to differentiate from the `submit_system_state` [warning](https://github.com/DataDog/integrations-core/pull/10783/files#diff-14804510c6025e372a3c4ddc9b506625f40a5a5f4c301d8666e21818df35a35eR81)?

I.e. "Encountered error getting Silk system state", "Encountered error getting Silk server state".